### PR TITLE
PR for Video listing with Filter

### PR DESCRIPTION
**Pull request by gopalbbharadva**
#### Features
- Filter by category
- Routing to all pages.
- Toggle option menu

#### Components:
- Sidebar
- Chips
- Video card
- Channel avatar

Live link: https://deploy-preview-2--elaborate-hotteok-315b5b.netlify.app/

### CODE REVIEW

File: src/frontend/contexts/FilterContext.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const FilterContext = createContext();
+
+const initialState = {
+  category: "All",
+};
+
+const FilterProvider = ({ children }) => {
+  const [filterState, filterDispatch] = useReducer(filterReducer, initialState);
+
+  console.log(filterState);
```
Comment (horsemaker) [inline src/frontend/contexts/FilterContext.jsx]: You might want to remove this `console.log`

File: src/frontend/helperfunctions/filterByCategory.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const filterByCategory = (categoryName, filterDispatch) => {
+  console.log("inside method", categoryName);
```
Comment (horsemaker) [inline src/frontend/helperfunctions/filterByCategory.js]: `console.log` here too

File: src/frontend/pages/videolist/VideolistPage.jsx [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
 
 export const VideolistPage = () => {
+  const { dataStoreState } = useDataStore();
+  const { categories, videos } = dataStoreState;
+  const { filterState } = useFilter();
+  const { category } = filterState;
+
+  const filteredVideos = videos.filter((item) => item.category === category);
+
+  console.log(filteredVideos);
```
Comment (horsemaker) [inline src/frontend/pages/videolist/VideolistPage.jsx]: `console.log`

**Review by horsemaker — APPROVED** [review]
Not gonna lie. The code looked really really clean & structured! Kudos for that! 👌

**Review by manojkumar8400 — APPROVED** [review]
Good to go.

**Comment by gopalbbharadva** [comment]
> Not gonna lie. The code looked really really clean & structured! Kudos for that! ok_hand

Thanks, I learned all my mistakes from e-commerce and trying to not repeat in VL.

**Pull request merged**